### PR TITLE
Update to ci scripts

### PR DESCRIPTION
**Pull request by adangel**
* Upload the new documentation during release to http://pmd.sourceforge.net/pmd-VERSION/ - this web space serves as an archive for the documentation
* Add nice ansi colors to the build log
* Don't fail the build, if upload of binaries/documentation to sourceforge fails (during snapshot build or release build). Note: This means, that travis will be green, even if the release didn't end up on sourceforge... so, this will probably be a manual step after the release, to check, whether the binaries/documentation is, where it is supposed to be.

### CODE REVIEW

File: .travis/build-coveralls.sh [hunk 1]
```diff
@@ -2,9 +2,10 @@
 set -e
 
 source .travis/common-functions.sh
+source .travis/colors.sh
 
 VERSION=$(./mvnw -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:1.5.0:exec | tail -1)
-echo "Building PMD Coveralls.io report ${VERSION} on branch ${TRAVIS_BRANCH}"
+echo_yellow "[ INFO] Building PMD Coveralls.io report ${VERSION} on branch ${TRAVIS_BRANCH}"
```
Comment (oowekyala) [inline .travis/build-coveralls.sh]: I think you should encapsulate the `[ INFO]` prefix + yellow color into a function. Same for errors, you could have `echo_info`, `echo_error`, etc.

File: .travis/release.sh [hunk 2]
```diff
@@ -9,10 +11,22 @@ if [ "${BUILD}" = "deploy" ]; then
 
 # Deploy to ossrh has already been done with the usual build. See build-deploy.sh
 
-# The site has been built before, the files have already been uploaded to sourceforge.
-# Since this is a release, making the binary the new default file...
-curl -H "Accept: application/json" -X PUT -d "default=windows&default=mac&default=linux&default=bsd&default=solaris&default=others" \
-     -d "api_key=${PMD_SF_APIKEY}" https://sourceforge.net/projects/pmd/files/pmd/${RELEASE_VERSION}/pmd-bin-${RELEASE_VERSION}.zip
+(
+    # disable fast fail, exit immediately, in this subshell
+    set +e
```
Comment (oowekyala) [inline .travis/release.sh]: If I understand right, this means that if curl returns a non-zero code, then the subshell doesn't immediately end and you can output your debug message. But when does the subshell terminate then?

File: .travis.yml [hunk 3]
```diff
@@ -52,11 +52,17 @@ deploy:
 after_deploy: bash .travis/release.sh
 
 notifications:
+  webhooks:
+    urls:
+      - https://webhooks.gitter.im/e/0e6dd310db54b96c2653
```
Comment (oowekyala) [inline .travis.yml]: Does this mean we'll get notifications on gitter during the build?

File: .travis/release.sh (hunk 2 above)
Comment (adangel) [reply .travis/release.sh]: The subshell ends with the closing parentheses  `)` . If one command fails within the subshell, then simply the next command is executed. Everytime you use parentheses, a new subshell is spawned.

File: .travis.yml (hunk 3 above)
Comment (adangel) [reply .travis.yml]: You should see this on the right on gitter (where currently activity on github is shown). I've added it to both gitter channels (pmd/pmd and pmd/maintainers)

File: .travis/build-coveralls.sh (hunk 1 above)
Comment (adangel) [reply .travis/build-coveralls.sh]: Will do

**Pull request merged**